k9mail/src/main/java/com/fsck/k9/widget/list/MessageListRemoteViewFactory.java

```diff
@@ -0,0 +1,208 @@
+import android.os.Binder;
+import android.support.v4.content.ContextCompat;
+import android.text.SpannableString;
+import android.text.style.StyleSpan;
+import android.view.View;
+import android.widget.RemoteViews;
+import android.widget.RemoteViewsService;
+
+import com.fsck.k9.K9;
+import com.fsck.k9.R;
+import com.fsck.k9.provider.MessageProvider;
+
+
+public class MessageListRemoteViewFactory implements RemoteViewsService.RemoteViewsFactory {
+    private static String[] MAIL_LIST_PROJECTIONS = {
+            MessageProvider.MessageColumns.SENDER,
+            MessageProvider.MessageColumns.SEND_DATE,
+            MessageProvider.MessageColumns.SUBJECT,
+            MessageProvider.MessageColumns.PREVIEW,
+            MessageProvider.MessageColumns.UNREAD,
+            MessageProvider.MessageColumns.HAS_ATTACHMENTS,
+            MessageProvider.MessageColumns.URI
+    };
+
+
+    private final Context context;
+    private final Calendar calendar;
+    private final ArrayList<MailItem> mailItems = new ArrayList<>(25);
+    private boolean senderAboveSubject;
+    private int readTextColor;
+    private int unreadTextColor;
+
+
+    public MessageListRemoteViewFactory(Context context) {
+        this.context = context;
+        calendar = Calendar.getInstance();
+    }
+
+    @Override
+    public void onCreate() {
+        senderAboveSubject = K9.messageListSenderAboveSubject();
+        readTextColor = ContextCompat.getColor(context, R.color.message_list_widget_text_read);
+        unreadTextColor = ContextCompat.getColor(context, R.color.message_list_widget_text_unread);
+    }
+
+    @Override
+    public void onDataSetChanged() {
+        long identityToken = Binder.clearCallingIdentity();
+        try {
+            loadMessageList();
+        } finally {
+            Binder.restoreCallingIdentity(identityToken);
+        }
+    }
+
+    private void loadMessageList() {
+        mailItems.clear();
+
+        Uri unifiedInboxUri = MessageProvider.CONTENT_URI.buildUpon().appendPath("inbox_messages").build();
+        Cursor cursor = context.getContentResolver().query(unifiedInboxUri, MAIL_LIST_PROJECTIONS, null, null, null);
+
+        if (cursor == null) {
+            return;
+        }
+
+        try {
+            while (cursor.moveToNext()) {
+                String sender = cursor.getString(0);
+                long date = cursor.isNull(1) ? 0L : cursor.getLong(1);
+                String subject = cursor.getString(2);
+                String preview = cursor.getString(3);
+                boolean unread = toBoolean(cursor.getString(4));
+                boolean hasAttachment = toBoolean(cursor.getString(5));
+                Uri viewUri = Uri.parse(cursor.getString(6));
+
+                mailItems.add(new MailItem(sender, date, subject, preview, unread, hasAttachment, viewUri));
+            }
+        } finally {
+            cursor.close();
+        }
+    }
+
+    @Override
+    public void onDestroy() {
+    }
```

Comment:
Oh, I see. My first thought for an overridden method with empty body was that it just didn't do whatever the super method does, I didn't see the class doesn't actually extend another.

`// Nothing to do here - method's just to satisfy the interface` could work, but not having anything is fine as well.